src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from text file and saving tasks in the file in Duke
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor for new instance.
+     *
+     * @param filePath path for text file to read and write from.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Read text file and obtain new array of tasks from it.
+     *
+     * @return new ArrayList of tasks.
+     * @throws DukeException error reading file.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        if (createFile()) {
+            try {
+                File f = new File(filePath);
+                Scanner s = new Scanner(f);
+                while (s.hasNext()) {
+                    String data = s.nextLine();
+                    String[] info = data.split("//");
+                    String type = info[0];
+                    String done = info[1];
+                    String name = info[2];
+                    if (type.equals("T")) {
+                        Task todo = new Todo(name);
+                        if (done.equals("1")) {
+                            todo.mark();
+                        }
+                        tasks.add(todo);
+                    } else if (type.equals("D")) {
+                        Task deadline = new Deadline(name, LocalDate.parse(info[3]));
+                        if (done.equals("1")) {
+                            deadline.mark();
+                        }
+                        tasks.add(deadline);
+                    } else if (type.equals("E")) {
+                        String date = info[3];
+                        String time = info[4];
+                        String dt = date + " " + time;
+                        Task event = new Event(name, LocalDateTime.parse(dt,
+                                DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")));
+                        if (done.equals("1")) {
+                            event.mark();
+                        }
+                        tasks.add(event);
+                    } else {
+                        throw new DukeException("Unknown Task type in text file");
+                    }
```

Comment:
Nice control flow and storage system, so I like the ideas behind it. However, I have a few suggestions to make regarding this structure.

There is far too many nested if statements. I suggest at least abstracting away the ```if(<task>.equals("<task>")``` blocks into a separate method in the task class itself, or maybe even in the ```Parser``` class as a static method. Another layer of abstraction possible is to separate the entire if block here into another method in this class, simply to reduce the levels of nested if statements and to reduce the size of this method.

The system is slightly too optimistic regarding the user. For example, if the user typos and accidentally has // in their task description.